paimon-core/src/main/java/org/apache/paimon/operation/LocalOrphanFilesClean.java

```diff
@@ -87,16 +90,27 @@ public LocalOrphanFilesClean(
                         table.coreOptions().deleteFileThreadNum(), "ORPHAN_FILES_CLEAN");
     }
 
-    public List<Path> clean() throws IOException, ExecutionException, InterruptedException {
+    public CleanOrphanFilesResult clean()
+            throws IOException, ExecutionException, InterruptedException {
         List<String> branches = validBranches();
 
         // specially handle to clear snapshot dir
-        cleanSnapshotDir(branches, deleteFiles::add);
+        cleanSnapshotDir(
+                branches,
+                deleteFiles::add,
+                p -> {
+                    try {
+                        deletedFilesSizeInBytes.addAndGet(fileIO.getFileSize(p));
```

Comment:
getFileSize should always be together with delete, otherwise, I feel there may be issues with execution efficiency, getFileSize should be executed parallelismly too.

Reply:
> getFileSize should always be together with delete, otherwise, I feel there may be issues with execution efficiency, getFileSize should be executed parallelismly too.

@JingsongLi Very thanks for your suggestions.

I would like to confirm with you the code improvement plan:
Are you means that getFileStatus execute only once in OrphanFilesClean#createFileCleaner, and fileCleaner can direct return the size of deleted file?

Reply:
I just think about it, maybe we don't need to invoke `getFileSize`, the files come from `listStatus`, it already contains file size.

Reply:
OK, I will have a try.
Thx

Reply:
> I just think about it, maybe we don't need to invoke `getFileSize`, the files come from `listStatus`, it already contains file size.

@JingsongLi 
Hi，please cc, THX.